crates/bevy_animation/src/lib.rs

```diff
@@ -123,9 +123,11 @@ impl VariableCurve {
         // An Ok(keyframe_index) result means an exact result was found by binary search
         // An Err result means the keyframe was not found, and the index is the keyframe
         // PERF: finding the current keyframe can be optimised
-        let search_result = self
-            .keyframe_timestamps
-            .binary_search_by(|probe| probe.partial_cmp(&seek_time).unwrap());
+        let search_result = self.keyframe_timestamps.binary_search_by(|probe| {
+            probe
+                .partial_cmp(&seek_time)
+                .expect("provided floats can't be compared")
```

Comment:
what are those floats? why are they compared? which is an input provided by the user when calling this function?